helm/cilium/templates/cilium-cni-configmap.yaml

```diff
@@ -0,0 +1,35 @@
+{{- if and (eq .Values.ipam.mode "eni") (.Values.cni.customConf) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cilium-cni-configuration
+  namespace: {{ .Release.Namespace }}
+data:
+  cni-config: |-
+      {
+        "cniVersion":"0.3.1",
```

Comment:
Do we need to watch out here? AFAIK there's already 0.4.0 as spec version (which is from 2019).

https://github.com/containernetworking/cni/releases/tag/spec-v0.4.0

Reply:
More information here: https://github.com/cilium/cilium/blob/v1.14.1/plugins/cilium-cni/main.go#L77

Reply:
Even on release 15 Cilium docs use this release https://docs.cilium.io/en/latest/network/concepts/ipam/eni/#create-a-cni-configuration

Haven't checked if it would support anything higher... I think there is even a release 1.0 or will be out there soon....

Reply:
Yeah for now it's fine I think. We only need to ensure we might not forget about it to update it once Cilium drops older CNI Versions

Reply:
Also it's the version they use in the default Cilium config https://github.com/cilium/cilium/blob/710316a4cf64202e210ec29c97be08209f96c69a/daemon/cmd/cni/config.go#L82